### PR TITLE
fix: [PL-39005]: misc fixes on the manager side

### PR DESCRIPTION
**Pull request by xingchi-jin**
## Describe your changes
1. fixed the sheduleTaskInternal function which will work for both init and execute api grpc handler
2. simplified taskClient interface. not to put uri and method in interface
3. Fix name of SchedulingTaskEvent class
4. move infra id into task payload when acquire. Because each individual task should know where the task locates. This helps for delegate in polling mode. 

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49837)

### CODE REVIEW

**Comment by mdmilic** [comment]
trigger uts

**Comment by xingchi-jin** [comment]
trigger uts

**Review by mdmilic — APPROVED** [review]
> Reviewed 13 of 13 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49837)* status: 14 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team. 

___
*[`400-rest/src/main/java/io/harness/grpc/DelegateServiceGrpcImpl.java` line 161 at r2](https://reviewable.io/reviews/harness/harness-core/49837#-NZ6eKLp9ZwBfYa4aM_v:-NZ6eKLqAtctwt90yfg0:boqwkai) ([raw file](https://github.com/harness/harness-core/blob/33cce6bdbdbd1ce609c3dcd90bddac7ab0871f44/400-rest/src/main/java/io/harness/grpc/DelegateServiceGrpcImpl.java#L161)):*
> ```Java
>         ExecutionInfraLocation locationInfo =
>             executionInfrastructureService.getExecutionInfrastructure(executionInfraRef.get());
>         capabilities.add(SelectorCapability.builder().selectors(Set.of(locationInfo.getDelegateGroupName())).build());
> ```

I think this is OK for now, but it's too restrictive for the general case. E.g. in K8S this is not needed and might not be needed for other use case. We could create some new type of check/selector capability which checks the infraRef in a more appropriate way.

**Comment by xingchi-jin** [comment]
trigger uts

**Comment by xingchi-jin** [comment]
trigger uts

File: 955-delegate-beans/src/main/proto/io/harness/delegate/core/beans/acquire_task.proto [hunk 1]
```diff
@@ -41,7 +40,8 @@ message TaskPayload {
   string resourceMethod = 3;
   string log_token =
       6;  // Token for log stream. It's not in the payload data because log token is assigned when acquire
-  InputData task_data = 7;   // binary data for task inputs.
-  InputData infra_data = 8;  // binary data for infra config
+  InputData task_data = 7;      // binary data for task inputs.
+  InputData infra_data = 8;     // binary data for infra config
+  string executionInfraId = 9;  // Id for the execution infra to use
```
Comment (gn-harness) [inline 955-delegate-beans/src/main/proto/io/harness/delegate/core/beans/acquire_task.proto:45]: how does delegate use this Id information?

**Review by gn-harness — APPROVED** [review]

File: 955-delegate-beans/src/main/proto/io/harness/delegate/core/beans/acquire_task.proto (hunk 1 above)
Comment (xingchi-jin) [reply 955-delegate-beans/src/main/proto/io/harness/delegate/core/beans/acquire_task.proto:45]: In init, this id is used to name the resources that are to be created.
In execute, this id is used to locate where to execute the task (eg, get le endpoint )

**Pull request merged**